listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -454,6 +454,105 @@ def fetch_recent_listens_for_users(self, users, min_ts: datetime = None, max_ts:
                 ))
         return listens
 
+
+    def fetch_recent_listens_for_users_new(self, users, min_ts: datetime, max_ts: datetime, limit=25):
+        """ Fetch recent listens for a list of users.
+
+            users: A list containing the users for which you'd like to retrieve recent listens.
+            min_ts: Only return listens with listened_at after this timestamp, required.
+            max_ts: Only return listens with listened_at before this timestamp, required.
+        """
+
+        user_id_map = {user["id"]: user["musicbrainz_id"] for user in users}
+
+        args = {"user_ids": tuple(user_id_map.keys()), "limit": max(limit, 100)}
```

Comment:
The `max(limit, 100)` here seems wrong. You probably want to do a min instead, either way it should be in the caller not here.

Reply:
Oopzz will fix it

Reply:
Done

Reply:
Don't calculate min here, instead do it in the function calling this.

Reply:
Luckily its already being in validate_get_endpoint_params just had this as failsafe if the functions was used anywhere else. Done